### PR TITLE
test(robot): add rwx nfsOptions to robot test scaffolding

### PR DESCRIPTION
**Pull request by yangchiu**
#### Which issue(s) this PR fixes:

Issue https://github.com/longhorn/longhorn/issues/7639
PR https://github.com/longhorn/longhorn-tests/pull/1690

#### What this PR does / why we need it:

adding a new storage class and rwx volume with overridden mount options to an e2e test.

### CODE REVIEW

**Comment by yangchiu** [comment]
cc @longhorn/qa

File: e2e/libs/workload/statefulset.py [hunk 1]
```diff
@@ -10,7 +10,7 @@
 from utility.utility import logging
 
 
-def create_statefulset(statefulset_name, volume_type, option):
+def create_statefulset(statefulset_name, volume_type, sc_name):
     filepath = "./templates/workload/statefulset.yaml"
```
Comment (roger-ryao) [inline e2e/libs/workload/statefulset.py:14]: For this change, do we need to add `sc_name` the default value, similar to what was done in https://github.com/longhorn/longhorn-tests/pull/1904/files#diff-cb5abaea0420a63451de0d96179c6a7859721fccec1cbdcd80940c8920eba665R34?

Comment (yangchiu) [reply e2e/libs/workload/statefulset.py:14]: Not necessary. `create_statefulset` in `statefulset.py` is called by `create_statefulset` in `statefulset_keywords.py` which already specifies the default value. You can reference `create_volume` in `volume_keywords.py` which uses the same strategy.

**Review by roger-ryao — APPROVED** [review]
LGTM

**Pull request merged**